server/service/src/sync/api_v6/core.rs

```diff
@@ -0,0 +1,109 @@
+use crate::sync::{api::ParsingResponseError, SyncApiSettings};
+use reqwest::{Client, Response};
+use thiserror::Error;
+use url::{ParseError, Url};
+
+use super::*;
+
+// For now port_offset of 2
+static PORT_OFFST: u16 = 2;
+
+#[derive(Debug, Clone)]
+pub(crate) struct SyncApiV6 {
+    pub(crate) url: Url,
+    pub(crate) sync_v5_settings: SyncApiSettings,
+}
+
+#[derive(Error, Debug)]
+pub enum SyncApiV6CreatingError {
+    #[error("Cannot parse url while creating SyncApiV5 instance url: '{0}'")]
+    CannotParseSyncUrl(String, #[source] ParseError),
+    #[error("Error while creating SyncApiV6 instance")]
+    Other(#[source] anyhow::Error),
+}
+
+async fn response_or_err(
+    result: Result<Response, reqwest::Error>,
+) -> Result<SyncPullResponseV6, SyncApiErrorVariantV6> {
+    let response = match result {
+        Ok(result) => result,
+        Err(error) => {
+            if error.is_connect() {
+                return Err(SyncApiErrorVariantV6::ConnectionError(error));
+            } else {
+                return Err(SyncApiErrorVariantV6::Other(error.into()));
+            }
+        }
+    };
+
+    // Not checking for status, expecting 200 only, even if there is error
+    let response_text = response
+        .text()
+        .await
+        .map_err(ParsingResponseError::CannotGetTextResponse)?;
+
+    let result = serde_json::from_str(&response_text).map_err(|source| {
+        ParsingResponseError::ParseError {
+            source,
+            response_text,
+        }
+    })?;
+
+    Ok(result)
+}
+
+pub(crate) fn get_omsupply_central_url(sync_v5_url: &str) -> Result<Url, ParseError> {
+    let mut url = Url::parse(sync_v5_url)?;
+    // Unwrap is safe unless domain is not http or https, see port_or_know_default()
+    url.set_port(Some(url.port_or_known_default().unwrap() + PORT_OFFST))
```

Comment:
Yeah, this is hacky, I agree, but alternatives was quite a bit of work ? To be discussed in the carry over issues/question